include/envoy/http/codec.h

```diff
@@ -260,6 +283,16 @@ class ClientConnection : public virtual Connection {
    * @return StreamEncoder& supplies the encoder to write the request into.
    */
   virtual StreamEncoder& newStream(StreamDecoder& response_decoder) PURE;
+
+  /**
+   * Called when the connection goes over its high watermark.
```

Comment:
"backing" has no special meaning. Just trying to make it more clear from naming what is going on. Your proposal (either) sounds great to me!